### ISSUE TITLE
`NP_NULL_ON_SOME_PATH` verdict depends on null-check operand order (`x == null` vs `null == x`)

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4147]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Check name

`NP_NULL_ON_SOME_PATH`

### Description

The null-dataflow analysis behind `NP_NULL_ON_SOME_PATH` is sensitive to the operand order of a reference null check. A method whose null guards are written as `x == null` produces a warning that disappears when the same guards are rewritten Yoda-style as `null == x`. Reference equality is symmetric, so `x == null` and `null == x` are the same condition and the analysis should reach the same verdict on both. The only bytecode difference is the null-compare idiom (`getfield x; ifnonnull` vs `aconst_null; getfield x; if_acmpne`), which does not change the control-flow facts the detector reasons about. (In the reduced example below the `x == null` form actually emits a false positive that the `null == x` form avoids, so the inconsistency manifests here as an operand-order-dependent false positive; either way the two forms should agree.)

### Reproducer

```java
// BEFORE — `x == null`, 1 report on `return x.hashCode()`
public class GuaranteedFieldDereference {
    Object x;

    int test12() {
        if (x == null)
            System.out.println("x is null");
        if (x == null)
            throw new NullPointerException();
        else
            return x.hashCode();
    }
}
```

```java
// AFTER — Yoda `null == x`, 0 reports
public class GuaranteedFieldDereference {
    Object x;

    int test12() {
        if (null == x)
            System.out.println("x is null");
        if (null == x)
            throw new NullPointerException();
        else
            return x.hashCode();
    }
}
```

### Expected behavior

`NP_NULL_ON_SOME_PATH` should reach the same verdict on both forms, since `x == null` and `null == x` are identical conditions; the operand order of a reference equality test must not change the result of the null-dataflow analysis.

### Actual behavior

- BEFORE (`x == null`): 1 report — `Possible null pointer dereference of x in GuaranteedFieldDereference.test12()`.
- AFTER (`null == x`): 0 reports.
